compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIECombineStridedOps.cpp

```diff
@@ -71,10 +72,16 @@ struct CombineStridedOps
         return WalkResult::interrupt();
       });
 
-      uint8_t sourceMemspaceInt = npuDmaOp.getSourceMemorySpaceAsUInt();
-      uint8_t targetMemspaceInt = npuDmaOp.getTargetMemorySpaceAsUInt();
-      AMDAIE::DmaDimConfig dmaDimConfig(deviceModel, sourceMemspaceInt,
-                                        targetMemspaceInt);
+      std::optional<uint8_t> sourceMemspaceInt =
+          npuDmaOp.getSourceMemorySpaceAsUInt();
+      std::optional<uint8_t> targetMemspaceInt =
+          npuDmaOp.getTargetMemorySpaceAsUInt();
+      if (!sourceMemspaceInt || !targetMemspaceInt) {
+        return npuDmaOp.emitOpError()
```

Comment:
emitting errors from patterns, is there an alternative?

Reply:
Updated to notifyMatchFailure